CMakeLists.txt

```diff
@@ -0,0 +1,23 @@
+cmake_minimum_required(VERSION 3.15...3.26)
+project(${SKBUILD_PROJECT_NAME} LANGUAGES CXX)
+
+set(CMAKE_CXX_STANDARD 17 CACHE STRING "C++ version selection")
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+# Force macOS toolchains to use C++17
+set(CMAKE_OSX_DEPLOYMENT_TARGET 10.15)
```

Comment:
Not sure if this matters.

But my understanding is that MACOSX_DEPLOYMENT_TARGET="12.0" is the currently minimum supported version of Mac OS.

I had to set this for pyflct for example for everything to work on Mac OS.

Reply:
Hmm I ended up setting this very purposefully to be the lowest version possible that would fully support C++17. I didn't try higher. What didn't work if you set it lower than 12?

Reply:
Oh I think this was actually a brew issue for pyflct. Brew doesn't support the older version of mac os anymore, so it was complaining, so I bumped the version up.

For this, I don't think it will matter. So ignore me.